src/Core/Localization/Currency/CurrencyDataSource.php

```diff
@@ -88,17 +82,29 @@ public function isCurrencyAvailable($currencyCode)
     }
 
     /**
-     * Get all the available (installed + active) currencies' data.
-     *
+     * {@inheritdoc}
+     */
+    public function getAvailableCurrenciesData($localeCode)
+    {
+        return $this->formatCurrenciesData($this->installedDataLayer->getAvailableCurrencyCodes(), $localeCode);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAllCurrenciesDataInDatabase($localeCode)
+    {
+        return $this->formatCurrenciesData($this->installedDataLayer->getAllCurrencyCodesInDatabase(), $localeCode);
+    }
+
+    /**
+     * @param array $currencyCodes
      * @param string $localeCode
-     *                           IETF tag. Data will be translated in this language
      *
-     * @return CurrencyData[]
-     *                        The available currencies' data
+     * @return array
      */
-    public function getAvailableCurrenciesData($localeCode)
+    private function formatCurrenciesData(array $currencyCodes, $localeCode)
```

Comment:
It's a BC, do we accept this?

Reply:
The BC here is the added method, same problem as the previous comment